File: Hackerrank/Compare-the-Triplets-2.py
# Problem link: https://www.hackerrank.com/challenges/compare-the-triplets/problem
# A resposta do probelma é a criação de uma função, que compare os elementes de dois array
# e segunindo algumas regras é gerado um novo array

#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the compareTriplets function below.
def compareTriplets(a, b):
    i=[0,0]
    for elem in range(3):
        if a[elem] > b[elem]:
            i[0] += 1
        elif a[elem] < b[elem]:
              i[1] += 1
    return i

a = [3,85,2]
b = [10,2,50]
print("Teste-1")
print(compareTriplets(a,b))

a = [10,5,2]
b = [10,2,50]
print("Teste-2")
print(compareTriplets(a,b))

File: Hackerrank/compare-the-triplets_2.cpp
#include <bits/stdc++.h>

using namespace std;

// complete the compareTriplets function below.
vector<int> compareTriplets(vector<int> a, vector<int> b){

    vector<int> a_b_scores(2,0);

    for(int i = 0; i < a.size(); ++i)
    {
        if(a[i] > b[i])
        {
            a_b_scores[0] = ++ a_b_scores[0];
        }
        if(a[i] < b[i])
        {
            a_b_scores[1] = ++ a_b_scores[1];
        }

        return a_b_scores;
    }

}